### ISSUE TITLE
[epic] False Positive tracking / triaging / categorization

### ISSUE THREAD

**Issue by PartiallyTyped** [rust-lang/rust-clippy#12046]
### Description

The aim of this issue is to track effort to reduce False Positive rate, and potentially categorize things into common patterns.

Obviously there are many more, and I will keep adding them as I go.

- [ ] #1693 | `while_let_loop`
- [ ] #1725 | `get_unwrap`
- [ ] #1089 | `useless_if_let_seq`
- [ ] #2042 | `op_ref`

## Could be closed(?)
- [ ] #488 (@blyxyas)
- [ ] #2025 (@Jarcho) 

## Have PR
- [ ] #2834 | `float_cmp` | #11948 (@Jarcho)

## Needs MVCE
- [ ] #1218

## Macros
This category is concerned with FPs that involve macros, and their expansions.

- [ ] #12037 | 
- [ ] #12035 | 
- [ ] #11274 | `redundant_closure_call` (duplicate of #1553 ?).
- [ ] #1553  | `redundant_closure_call`
- [ ] #2373  | `unnecessary_mut_passed`
- [ ] #2373  | `zero_ptr`

## May need discussion 

- [ ] #488 | `eq_op`

### Version

```text
N/A
```

**Comment by PartiallyTyped** [comment]
The `boxed_local` ones need a minor change on their lint message and are good to go.

**Comment by blyxyas** [comment]
Done! ☑ =^w^=
